### PR TITLE
feat(asset tree): don't show "expand" arrow on childless assets

### PR DESCRIPTION
**Pull request by polomani**
https://github.com/cognitedata/gearbox.js/issues/454
I also did some refactoring on component, as simply changing `isLeaf=true` didn't work.
So I started digging and realised that the structures we use are not very "reactive".
That's why I decided to use just one map "assetId => node" in the state. This makes all the state operations obvious, reactive and always top-down (not updating assetTree from bottom `node.refData.props.whatever = value`)

### CODE REVIEW

**Comment by polomani** [comment]
I also removed tree creation on root assets, why on earth we are doing that if api already provides us only root assets?

**Comment by cognite-cicd** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-461.gearboxjs.preview.cogniteapp.com

**Comment by cognite-cicd** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-461.gearboxjs.preview.cogniteapp.com

File: src/components/AssetTree/AssetTree.tsx [hunk 1]
```diff
@@ -46,45 +45,17 @@ class AssetTree extends React.Component<AssetTreeProps, AssetTreeState> {
     theme: { ...defaultTheme },
   };
 
-  static mapDataAssets(assets: Asset[]): AssetTreeNode[] {
+  static convertAssetsToNodeMap(assets: Asset[]) {
```
Comment (mrtraser) [inline src/components/AssetTree/AssetTree.tsx]: Provide return type here, pls.

File: src/components/AssetTree/AssetTree.tsx [hunk 2]
```diff
@@ -135,37 +108,37 @@ class AssetTree extends React.Component<AssetTreeProps, AssetTreeState> {
...
-    }
     const eventKey = treeNode.props.eventKey;
     const assetId = eventKey ? Number.parseInt(eventKey, 10) : undefined;
-    if (!(assetId && !Number.isNaN(assetId))) {
-      return;
-    }
+    if (!treeNode.props.children && assetId) {
+      const updatedAssetNode = { ...this.state.assetNodesMap[assetId] };
+      const assetChildren = await this.cursorApiRequest(assetId);
+      let mapUpdate: { [s: number]: AssetTreeNode };
```
Comment (mrtraser) [inline src/components/AssetTree/AssetTree.tsx]: Looks like type mismatch, should be `{[id: string]: AssetTreeNode}`

**Comment by cognite-cicd** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-461.gearboxjs.preview.cogniteapp.com

**Pull request merged**